File: app/src/main/java/yoyo/app/android/com/yoyoapp/DataModels/Location.java
package yoyo.app.android.com.yoyoapp.DataModels;

public class Location {
    private String locationId;
    private String code;
    private String title;
    private int order;
    private Double lat;
    private Double lon;

    public String getLocationId() {
        return locationId;
    }

    public void setLocationId(String locationId) {
        this.locationId = locationId;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }


    public int getOrder() {
        return order;
    }

    public void setOrder(int order) {
        this.order = order;
    }

    public Double getLon() {
        return lon;
    }

    public void setLon(Double lon) {
        this.lon = lon;
    }

    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }
}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/DataModels/Location.kt
package yoyo.app.android.com.yoyoapp.DataModels

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import ir.mirrajabi.searchdialog.core.Searchable

@Entity(tableName = "locations")
data class Location(@PrimaryKey
                    @NonNull
                    var code: String = "",
                    @ColumnInfo(name = "title")
                    var name: String? = null,
                    var order: Int = 0,
                    var lat: Double? = null,
                    var lon: Double? = null):Searchable {

    override fun getTitle(): String {
        return name ?: ""
    }
}
